docs/python/1.6/python-appapi-devguide-5.md

```diff
@@ -67,3 +77,181 @@ def main():
 ~~~~~~
 
 Any transforms that are performed on the parallelized `Stream` object occur in parallel to the degree that is specified in the `.parallel()` function. In this example, you specified 4, which means that four channels process the data in the parallel region on the graph.
+
+
+<a id="consistent_region"></a>
+
+# Consistent region
+
+Because of business requirements, some applications require that all tuples in an application are processed at least once.
+By default any state is reset to its initial state after a processing element (PE) restart. A restart may occur due to:
+
+* a failure in the PE or its resource,
+* a explicit PE restart request,
+* a parallel region width change (IBM Streams 4.3 or later)
+
+You can use a consistent region in your stream processing applications to avoid data loss due to software or hardware failure and meet your requirements for at-least-once processing.
+
+The application or a portion of it may be configured to maintain state after a PE restart by the mechanism **consistent region**.
+A **consistent region** is a subgraph where the states of callables become consistent by processing all the tuples at least once within defined points on a stream.
+On a failure, a region is reset to its last successfully persisted state, and source callable of a region can replay any tuples submitted after the restored state.
+
+![cr_reset](/streamsx.documentation/images/python/cr_reset.png)
+
+
+The consistent region feature requires a configured [checkpoint repository](https://www.ibm.com/support/knowledgecenter/SSCRJU_5.3/com.ibm.streams.cfg.doc/doc/ibminfospherestreams-configuring-checkpoint-data-store.html) in the Streams instance and Streams applications must be submitted in [DISTRIBUTED](https://streamsxtopology.readthedocs.io/en/stable/streamsx.topology.context.html#streamsx.topology.context.ContextTypes.DISTRIBUTED) context.
+
+This section will discuss how to create stream processing applications with the consistent region feature:
+
+* [Configure a consistent region](#configure)
+* [Stateful callables](#stateful)
+* [Use SPL operators supporting consistent region](#spl_cr)
+
+
+<a id="configure"></a>
+## Configure a consistent region
+
+* [streamsx.topology.topology.Stream.set_consistent()](https://streamsxtopology.readthedocs.io/en/stable/streamsx.topology.topology.html#streamsx.topology.topology.Stream.set_consistent)
+* [ConsistentRegionConfig](https://streamsxtopology.readthedocs.io/en/stable/streamsx.topology.state.html#streamsx.topology.state.ConsistentRegionConfig)
+
+The recommended way to create a `ConsistentRegionConfig` is to call either `operator_driven()` or `periodic()`.
+
+For example, set a source stream `s` to be a the start of an operator driven consistent region with a drain timeout of five seconds and a reset timeout of twenty seconds:
+```
```

Comment:
Drain and reset timeouts of 5 or 20 seconds are no good examples. Readers might think, that these values are good values, which fit the average streaming application in most cases. The default timeouts for drain/reset are 180 seconds, I think. This example should go to higher timeouts, at least 30 or 60 seconds.